teps/0138-improve-tekton-pipeline-feature-flags.md

```diff
@@ -0,0 +1,73 @@
+
+- For maintainers, the maintenance operation of swapping the storage version from v1beta1 to v1 should not have affected our users. However, we had to [change the user-facing default value of enable-api-fields from `stable` to `beta` ](https://github.com/tektoncd/pipeline/pull/6732) before changing the storage version of the API to [avoid breaking PipelineRuns using `beta` features](https://github.com/tektoncd/pipeline/pull/6444#issuecomment-1580926707).
+
+- When promoting features, it could cause confusions for contributors to be dependent on the fact whether an apiVersion is available. For example, during [the promotion to beta for projected workspaces](https://github.com/tektoncd/pipeline/pull/5530), the v1 api's existence led to confusions of what to do with beta features in v1beta1 and its difference with in v1.
+
+## Goals
+
+- [Decouple API versioning and feature stability](https://github.com/tektoncd/pipeline/issues/6592)
+  - Feature validation and implementation should be the same for all API versions.
+- Cluster operators can enable an individual feature without being forced to opt into other features at the same stability level.
+  - Increase the visibility of stability levels for features that have their own feature flags and ones that are not tied with the api spec
+- Describe a testing strategy that will give us confidence in our implementation
+  - One of the main reasons why [TEP-0033](https://github.com/tektoncd/community/blob/main/teps/0033-tekton-feature-gates.md#pros-and-cons) chose the existing style of feature flags was to have a simpler testing matrix.  With upgrades to the feature versioning, we need to ensure that our testing strategy gives us confidence in our feature coverage.
+  - Restructure conditions for integration tests to avoid  [Skipped Integration Tests](https://github.com/tektoncd/pipeline/issues/6079) based on different required sets of behavioral feature flags in certain stability levels
+- Minimize backwards incompatible changes and provide simple migration paths for cluster operators.
+- Cluster operators can set a default feature stability level.
+  - For example, we want to address the current case where a cluster operator only wants to use `stable` features and we have defaulted `beta` as the opt-in features stability then they would have to change back the default feature stability to use all `stable` features by reading through the release notes.
+- Build consistency in the usage for the group feature flag and the behavioral feature flags.
+
+## Non goals
+
+- Better guidance on feature promotion and when features can be promoted
```

Comment:
I agree. 
I have the feeling that the lack of guidance has to stagnating features and consequent issues when introducing new  API versions, but definitely the root issue there is the coupling of API and feature versions.
I took an action item to define this guidance, which I have not done yet, I will propose something soon.